show-line-of-sight-between-geoelements/src/main/java/com/esri/arcgismaps/sample/showlineofsightbetweengeoelements/components/SceneViewModel.kt

```diff
@@ -0,0 +1,259 @@
+
+        // Define base surface for elevation data
+        val surface = Surface().apply {
+            elevationSources.add(
+                ArcGISTiledElevationSource(
+                    uri = getString(
+                        application,
+                        R.string.elevation_service_url
+                    )
+                )
+            )
+        }
+
+        // Define a scene layer for the New York buildings
+        val buildings =
+            ArcGISSceneLayer(uri = application.getString(R.string.new_york_buildings_service_url))
+
+        // Add the surface and buildings to the scene, and define the viewpoint on launch
+        scene.apply {
+            baseSurface = surface
+            operationalLayers.add(buildings)
+            initialViewpoint = Viewpoint(
+                boundingGeometry = observerGraphic.geometry as Point,
+                camera = camera
+            )
+        }
+
+        // Set up a heading expression to handle graphic rotation
+        val renderer3D = SimpleRenderer().apply {
+            sceneProperties.headingExpression = ("[HEADING]")
+        }
+
+        // Add the graphics to the graphic overlay
+        graphicsOverlay.apply {
+            sceneProperties.surfacePlacement = SurfacePlacement.RelativeToScene
+            graphicsOverlay.graphics.add(observerGraphic)
+            graphicsOverlay.graphics.add(taxiGraphic)
+            graphicsOverlay.renderer = renderer3D
+        }
+
+        viewModelScope.launch {
+
+            scene.load().onSuccess {
+
+                // Create a line of sight between the two graphics and add it to the analysis overlay
+                val lineOfSight = GeoElementLineOfSight(
+                    observerGeoElement = observerGraphic,
+                    targetGeoElement = taxiGraphic
+                )
+                analysisOverlay.analyses.add(lineOfSight)
+
+                // Select (highlight) the taxi when the line of sight target visibility changes to visible
+                if (lineOfSight.isVisible) {
+                    taxiGraphic.isSelected = lineOfSight.isVisible
+                }
+
+                // Create a timer to animate the tank
+                val timer = Timer()
+                timer.schedule(object : TimerTask() {
+                    override fun run() {
+                        animate()
+                    }
+                }, 0, 50)
+
+            }.onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    title = "Scene Failed to Load",
+                    description = error.cause.toString()
+                )
+
+            }
+
+
+        }
+
+    }
+
+    /**
+     * Updates [height] of the Z value
+     */
+    fun updateHeight(height: Double) {
+        val pointBuilder = PointBuilder(observerGraphic.geometry as Point).apply {
+            z = height
+        }
+        observerGraphic.geometry = pointBuilder.toGeometry()
+        _currentZValue.value = height
```

Comment:
Could you rename this stateflow to `_observerHeight` and `observerHeight`
```suggestion
        _observerHeight.value = height
```